test/components/splitters/test_hierarchical_doc_splitter.py

```diff
@@ -0,0 +1,162 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+
+from haystack import Document, Pipeline
+from haystack_experimental.components.splitters import HierarchicalDocumentSplitter
+from haystack.components.writers import DocumentWriter
+from haystack.document_stores.in_memory import InMemoryDocumentStore
+
+
+class TestHierarchicalDocumentSplitter:
+    def test_init_with_default_params(self):
+        builder = HierarchicalDocumentSplitter(block_sizes=[100, 200, 300])
+        assert builder.block_sizes == [300, 200, 100]
+        assert builder.split_overlap == 0
+        assert builder.split_by == "word"
+
+    def test_init_with_custom_params(self):
+        builder = HierarchicalDocumentSplitter(block_sizes=[100, 200, 300], split_overlap=25, split_by="word")
+        assert builder.block_sizes == [300, 200, 100]
+        assert builder.split_overlap == 25
+        assert builder.split_by == "word"
+
+    def test_init_with_duplicate_block_sizes(self):
+        try:
+            HierarchicalDocumentSplitter(block_sizes=[100, 200, 200])
+        except ValueError as e:
+            assert str(e) == "block_sizes must not contain duplicates"
```

Comment:
```suggestion
        with pytest.raises(ValueError):
            HierarchicalDocumentSplitter(block_sizes=[100, 200, 200])
```

we usually do not capture/check error messages